### PR TITLE
[SPARK-13457][SQL] Removes DataFrame RDD operations

### PR DESCRIPTION
**Pull request by liancheng**
## What changes were proposed in this pull request?

This PR removes DataFrame RDD operations. Original calls are now replaced by calls to methods of `DataFrame.rdd`.
## How was the this patch tested?

No extra tests are added. Existing tests should do the work.

### CODE REVIEW

File: examples/src/main/scala/org/apache/spark/examples/ml/DecisionTreeExample.scala [hunk 1]
```diff
@@ -324,6 +324,7 @@ object DecisionTreeExample {
 
   /**
    * Evaluate the given RegressionModel on data.  Print the results.
+ *
```
Comment (mengxr) [inline examples/src/main/scala/org/apache/spark/examples/ml/DecisionTreeExample.scala]: Fix indentation.

File: examples/src/main/scala/org/apache/spark/examples/ml/OneVsRestExample.scala [hunk 2]
```diff
@@ -155,7 +155,7 @@ object OneVsRestExample {
 
     // evaluate the model
     val predictionsAndLabels = predictions.select("prediction", "label")
-      .map(row => (row.getDouble(0), row.getDouble(1)))
+      .rdd.map(row => (row.getDouble(0), row.getDouble(1)))
```
Comment (mengxr) [inline examples/src/main/scala/org/apache/spark/examples/ml/OneVsRestExample.scala]: we should either put `.rdd` at the end of the previous line or make a new line for it

**Comment by rxin** [comment]
LGTM pending tests pass. 

And address Xiangrui's comment too.

File: examples/src/main/scala/org/apache/spark/examples/ml/OneVsRestExample.scala (hunk 2 above)
Comment (liancheng) [inline examples/src/main/scala/org/apache/spark/examples/ml/OneVsRestExample.scala]: Thanks, I did the change using `sed`. Will go through all the changes to fix similar occurrences.

**Comment by liancheng** [comment]
Thanks for the review, addressed those styling issues.

**Comment by liancheng** [comment]
retest this please

**Comment by liancheng** [comment]
The build failure was introduced by newly merged MLlib changes in master. Rebased to fix it.

**Comment by liancheng** [comment]
retest this please

**Comment by liancheng** [comment]
Failed test case in the last two build failures is "SQLListenerSuite.SPARK-11126: no memory leak when running non SQL jobs", which is irrelevant to this PR. I'm going to merge this to master.

**Comment by davies** [comment]
@liancheng The failed test is actually introduced by this PR, so I'm going to revert this patch.

**Comment by liancheng** [comment]
@davies Thanks a lot and sorry for the trouble! `DataFrame.foreach` and `DataFrame.foreachPartitions` shouldn't be removed since they are actions. The failure was because the `withNewExecutionId` calls were removed together with these two methods and thus Spark jobs are not properly tracked.

Opened #11388 for another try.

**Pull request closed without merging**